Add pinmap docs to target porting summary page

c1728p9 opened the pull request:
Add information on creating pinmaps for new targets in the porting page.

**NOTE**: THIS IS ONLY FOR 5.12. PLEASE DON'T MERGE INTO 5.11.

c1728p9 committed:
Add pinmap docs to target porting summary page

Add information on creating pinmaps for new targets in the porting
page.

AnotherButler commented:
@c1728p9 It looks like at least one code snippet is failing. Please look into this.

c1728p9 commented:
Updated these docs to reflect changes in #9449

AnotherButler commented:
ping @donatieng

donatieng reviewed (approved):
Looks good!

Commit:
Edit PinMap addition to target.md

Edit new content about PinMap, mostly for active voice and consistent spelling.

AnotherButler reviewed (commented):
I've edited this content, mostly for active voice. Please make sure I didn't accidentally change the meaning of anything.
